packages/components/organisms/f-checkout/src/store/tests/analytics.module.test.js

```diff
@@ -0,0 +1,391 @@
+                    mapAnalyticsNamesSpy.mockImplementation(() => field);
+                });
+
+                it(`should call ${UPDATE_AUTOFILL} with correct field`, () => {
+                    // Act
+                    updateAutofill({ commit }, checkoutState);
+
+                    // Assert
+                    expect(commit).toHaveBeenCalledWith(UPDATE_AUTOFILL, autofillFields);
+                });
+
+                it('should call `mapAnalyticsNames` with correct field', () => {
+                    // Act
+                    updateAutofill({ commit }, checkoutState);
+
+                    // Assert
+                    expect(commit).toHaveBeenCalledWith(UPDATE_AUTOFILL, autofillFields);
+                });
+            });
+        });
+
+        describe('updateChangedField ::', () => {
+            const field = 'firstName';
+            let mapAnalyticsNameSpy;
+
+            beforeEach(() => {
+                mapAnalyticsNameSpy = jest.spyOn(mapper, 'mapAnalyticsName').mockImplementation(() => field);
+            });
+
+            it('should call `mapAnalyticsName` with passed field', () => {
+                // Act
+                updateChangedField({ commit }, field);
+
+                // Assert
+                expect(mapAnalyticsNameSpy).toHaveBeenCalledWith(field);
+            });
+
+            it(`should call ${UPDATE_CHANGED_FIELD} with passed field`, () => {
+                // Act
+                updateChangedField({ commit }, field);
+
+                // Assert
+                expect(commit).toHaveBeenCalledWith(UPDATE_CHANGED_FIELD, field);
+            });
+        });
+
+        describe('trackInitialLoad ::', () => {
+            const expectedEvent = {
+                custom: {
+                    checkout: {
+                        step: 1
+                    },
+                    basket: {
+                        id: rootState.checkout.basket.id,
+                        total: rootState.checkout.basket.total
+                    },
+                    restaurant: {
+                        id: rootState.checkout.restaurantId
+                    },
+                    pageData: {
+                        name: 'Checkout 1 Guest',
+                        group: 'Checkout'
+                    }
+                }
+            };
+
+            it('should call `event` method of `f-trak`', () => {
+                // Act
+                trackInitialLoad({ rootState, dispatch });
+
+                // Assert
+                expect(trackEventSpy).toHaveBeenCalledWith(expectedEvent);
+            });
+
+            describe('when user `isLoggedIn` is true', () => {
+                it('should set `name` to `Checkout 1 Overview`', () => {
+                    // Arrange
+                    const expectedName = 'Checkout 1 Overview';
+
+                    rootState.checkout.isLoggedIn = true;
+                    expectedEvent.custom.pageData.name = expectedName;
```

Comment:
I would probably create a new object and override the value on the new object, otherwise you could have a situation where one test affects another.

Reply:
You could maybe call the default one `baseEvent`, and then create a new variable inside each test called `expectedEvent` which you assert against.

Reply:
I've created a `beforeEach` for this so hopefully will prevent any issues